src/utils/auth.js

```diff
@@ -0,0 +1,35 @@
+import checkResponse from './utils';
+import { baseUrl, headers } from './serverApi';
+
+export function signup(data) {
+  return fetch(`${baseUrl}/signup`, {
+    method: 'POST',
+    headers: headers,
+    body: JSON.stringify({
+      name: data.name,
+      email: data.email,
+      password: data.password,
+    }),
+  }).then((res) => checkResponse(res));
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.